### PR TITLE
CSEN-17 WeightedQueue was added with a test report

### PR DESCRIPTION
**Pull request by MedvedevBW**

### CODE REVIEW

**Comment by MedvedevBW** [comment]
[run tests]

**Comment by MedvedevBW** [comment]
[run tests]

**Comment by MedvedevBW** [comment]
@placccebo

File: src/main/scala/com/bwsw/cloudstack/entities/common/WeightedQueue.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import com.bwsw.cloudstack.entities.common.traits.Queue
+import com.google.common.collect.{Range, RangeMap, TreeRangeMap}
+
+import scala.util.Random
+
+/**
+  * Class is responsible for obtaining a list item with a probability calculated by the formula:
+  * (length - indexOfElement)/sumOfAllNumbersOfList
+  * where:
+  * "length" - length of initialization list with elements
```
Comment (placccebo) [inline src/main/scala/com/bwsw/cloudstack/entities/common/WeightedQueue.scala]: use x,y,z for formula presentation

**Review by placccebo — CHANGES_REQUESTED** [review]

File: src/main/scala/com/bwsw/cloudstack/entities/common/WeightedQueue.scala (hunk 1 above)
Comment (MedvedevBW) [reply src/main/scala/com/bwsw/cloudstack/entities/common/WeightedQueue.scala]: ok

**Comment by MedvedevBW** [comment]
@placccebo

**Review by placccebo — APPROVED** [review]

**Pull request merged**